src/Riok.Mapperly/Descriptors/Mappings/UserMappings/UserImplementedExistingTargetMethodMapping.cs

```diff
@@ -32,6 +32,24 @@ public override IEnumerable<StatementSyntax> Build(TypeMappingBuildContext ctx,
         // we explicitly cast to be able to use the default interface implementation or explicit implementations
         if (Method.ReceiverType?.TypeKind != TypeKind.Interface)
         {
+            // if target parameter expects the ref keyword we need to add some extra lines so the reference is correctly stored back into the property.
+            if (targetParameter.RefKind == RefKind.Ref)
+            {
+                var targetRefVarName = Method.Name + "TargetRef";
```

Comment:
My bad, `Method.Name` doesn’t really make sense in this context. I had confused it with the name of the target field. Since the target name isn’t easily accessible, I’d simply use
```csharp
var targetRefVarName = ctx.NameBuilder.New("targetRef");
```